CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+# Contribute
+
+Want to show Sinatra some love? Help out by contributing!
+
+## Find a bug?
+
+Log it in our [issue tracker][ghi] or send a
+note to the [mailing list][ml].
+Be sure to include all relevant information, like the versions of Sinatra and
+Ruby you're using. A [gist](http://gist.github.com/) of the code that caused
+the issue as well as any error messages are also very helpful.
+
+## Need help?
+
+The [Sinatra mailing list][https://groups.google.com/forum/#!forum/sinatrarb] has
+over 900 subscribers, many of which are happy to help out newbies or talk about
+potential feature additions. You can also drop by the
+[#sinatra](irc://chat.freenode.net/#sinatra) channel on
+[irc.freenode.net](http://freenode.net).
+
+## Have a patch?
+
+Bugs and feature requests that include patches are much more likely to
+get attention. Here are some guidelines that will help ensure your patch
+can be applied as quickly as possible:
+
+1. **Use [Git](http://git-scm.com) and [GitHub](http://github.com):**
+   The easiest way to get setup is to fork the
+   [sinatra/sinatra repo](http://github.com/sinatra/sinatra/).
+   Or, the [sinatra.github.com repo](http://github.com/sinatra/sinatra.github.com/),
+   if the patch is doc related.
+
+2. **Write unit tests:** If you add or modify functionality, it must
+   include unit tests. If you don't write tests, we have to, and this
+   can hold up acceptance of the patch.
+
+3. **Mind the `README`:** If the patch adds or modifies a major feature,
+   modify the `README.md` file to reflect that. Again, if you don't
+   update the `README`, we have to, and this holds up acceptance.
+
```

Comment:
I've added a step about the Changelog. I'd welcome suggestions for improvements to the phrasing here.